lib/common/util.js

```diff
@@ -337,13 +337,7 @@ function makeAuthenticatedRequestFactory(config) {
     }
 
     function onAuthenticated(err, authenticatedReqOpts) {
-      // google-auth-library returns a "Could not load..." error if it can't get
-      // an access token. However, it's possible an API request doesn't need to
-      // be authenticated, e.g. when downloading a file from a public bucket. We
-      // consider this error a warning, and allow the request to go through
-      // without authentication, relying on the upstream API to return an error
-      // the user would find more helpful, should one occur.
-      if (err && err.message.indexOf('Could not load') === -1) {
```

Comment:
So here we are, ignoring the "Could not load" error. The whole error handling block is skipped, but the bug is: the `authenticatedReqOpts` argument to this function is `undefined`. To fix this, we would just do: `authenticatedReqOpts = reqOpts` (the original reqOpts that we tried to authenticate. Since it failed, we would just use the unauthenticated ones)